### PR TITLE
Javascript Bundles

### PR DESCRIPTION
**Pull request by stefwalter**
Create javascript bundles that combine multiple AMD modules into a single file for quick loading.

### CODE REVIEW

**Comment by stefwalter** [comment]
Needs documentation.

File: tools/Makefile-tools.am [hunk 1]
```diff
@@ -1,7 +1,8 @@
 EXTRA_DIST += \
 	tools/html-minifier \
 	tools/htmlmin.conf \
-	tools/jsmerge \
+	tools/jsbundle \
+        tools/cssbundle \
```
Comment (mvollmer) [inline tools/Makefile-tools.am]: whitespace

**Comment by stefwalter** [comment]
Fixed whitespace, rebased.

**Comment by mvollmer** [comment]
The check-menu failure is genuine.  If anything goes wrong with dropdowns, it usually means that bootstrap is loaded twice.  Maybe that is the case?

**Comment by mvollmer** [comment]
When loading the dashboard (which I think avoids loading the "shell" component), I get this:

```
GET https://f22:9090/#/dashboard/list [HTTP/1.1 200 OK 20ms]
GET https://f22:9090/cockpit/$a2d150c685f4eb5d3382b34515b3a3882f263270/base1/cockpit.css [HTTP/1.1 304 Not Modified 1ms]
GET https://f22:9090/cockpit/$a2d150c685f4eb5d3382b34515b3a3882f263270/shell/shell.css [HTTP/1.1 304 Not Modified 1ms]
GET https://f22:9090/cockpit/$a2d150c685f4eb5d3382b34515b3a3882f263270/base1/bundle.js [HTTP/1.1 304 Not Modified 1ms]
GET https://f22:9090/cockpit/$a2d150c685f4eb5d3382b34515b3a3882f263270/shell/bundle.js [HTTP/1.1 304 Not Modified 1ms]
GET https://f22:9090/cockpit/$a2d150c685f4eb5d3382b34515b3a3882f263270/base1/patternfly.css [HTTP/1.1 304 Not Modified 1ms]
GET https://f22:9090/cockpit/$a2d150c685f4eb5d3382b34515b3a3882f263270/legacy/cockpit.css [HTTP/1.1 304 Not Modified 1ms]
GET https://f22:9090/cockpit/$a2d150c685f4eb5d3382b34515b3a3882f263270/base1/branding.css [HTTP/1.1 304 Not Modified 1ms]
GET https://f22:9090/cockpit/$a2d150c685f4eb5d3382b34515b3a3882f263270/legacy/patternfly.css [HTTP/1.1 304 Not Modified 1ms]
GET https://f22:9090/cockpit/$a2d150c685f4eb5d3382b34515b3a3882f263270/shell/images/server-small.png [HTTP/1.1 304 Not Modified 1ms]
GET https://f22:9090/cockpit/static/brand.png [HTTP/1.1 200 OK 42ms]
GET https://f22:9090/cockpit/$a2d150c685f4eb5d3382b34515b3a3882f263270/base1/jquery.js [HTTP/1.1 304 Not Modified 2ms]
GET https://f22:9090/cockpit/$a2d150c685f4eb5d3382b34515b3a3882f263270/base1/cockpit.js [HTTP/1.1 304 Not Modified 2ms]
GET https://f22:9090/cockpit/$a2d150c685f4eb5d3382b34515b3a3882f263270/manifests.js [HTTP/1.1 304 Not Modified 2ms]
GET https://f22:9090/cockpit/$a2d150c685f4eb5d3382b34515b3a3882f263270/base1/po.en.js [HTTP/1.1 304 Not Modified 1ms]
GET https://f22:9090/socket
```

Thus, both base1/bundle.js and base1/jquery.js are loaded (but I am not sure whether they are loaded into the same iframe).

**Comment by stefwalter** [comment]
> The check-menu failure is genuine. If anything goes wrong with dropdowns, it usually means that bootstrap is loaded twice. Maybe that is the case?

I'll take a look at this in a minute.

**Comment by stefwalter** [comment]
Fixed CI issue. It was because requirejs wasn't first in the base1 bundle. I broke this during cleanup of the patches.

Rebased as well.

**Pull request closed without merging**